polaris/experimental/_benchmark_v2.py

```diff
@@ -0,0 +1,247 @@
+from functools import cached_property
+from hashlib import md5
+from typing import Any, Callable, ClassVar, Generator, Literal, Sequence
+
+from loguru import logger
+from pydantic import BaseModel, ConfigDict, Field, computed_field, field_validator, model_validator
+from pydantic.alias_generators import to_camel
+from pyroaring import BitMap
+from typing_extensions import Self
+
+from polaris.benchmark import BenchmarkSpecification
+from polaris.benchmark._base import ColumnName
+from polaris.dataset import Subset
+from polaris.experimental._dataset_v2 import DatasetV2
+from polaris.utils.errors import InvalidBenchmarkError
+
+
+class IndexSet(BaseModel):
+    """
+    A set of indices for a split, either training or test.
+
+    It wraps a Roaring Bitmap object to store the indices, and provides
+    useful properties when serializing for upload to the Hub.
+    """
+
+    model_config = ConfigDict(arbitrary_types_allowed=True, alias_generator=to_camel)
+
+    indices: BitMap = Field(default_factory=BitMap, frozen=True, exclude=True)
+
+    @field_validator("indices", mode="before")
+    @classmethod
+    def _validate_indices(cls, v: BitMap | Sequence[int]) -> BitMap:
+        """
+        Accepts an initial sequence of ints, and turn it into a BitMap
+        """
+        if isinstance(v, BitMap):
+            return v
+        return BitMap(v)
+
+    @computed_field
+    @cached_property
+    def datapoints(self) -> int:
+        return len(self.indices)
```

Comment:
**Suggestion:** Minor, but I would rename to `n_datapoints`.

Reply:
I'm using this to match the the Hub-side data structure.